core-bundle/src/Filesystem/FilesystemItem.php

```diff
@@ -118,6 +120,40 @@ public function getPath(): string
         return $this->path;
     }
 
+    public function isVideo(): bool
+    {
+        return str_starts_with($this->getMimeType(), 'video/');
+    }
+
+    public function isAudio(): bool
+    {
+        return str_starts_with($this->getMimeType(), 'audio/');
+    }
+
+    public function isImage(): bool
+    {
+        return str_starts_with($this->getMimeType(), 'image/');
+    }
+
+    public function isPdf(): bool
+    {
+        return 'application/pdf' === $this->getMimeType();
+    }
+
+    public function isSpreadsheet(): bool
+    {
+        return \in_array(
+            $this->getMimeType(),
+            [
+                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+                'application/vnd.ms-excel',
+                'application/msexcel',
+                'application/x-msexcel',
+            ],
+            true,
+        );
+    }
```

Comment:
Just a nitpick with these new functions: If you call them on an item that is not a file, the logic exception will still say `"Cannot call getMimeType() on a non-file fileysytem item."` instead of `"Cannot call isSpreadsheet() …"`. Maybe we can improve this by always walking up the call-stack to the first function call of `FilesystemItem` in `assertIsFile()`? :thinking:

Reply:
I think they should just return `false` in this case. Implemented in 45f30a40d622cce743389f52f101e67232c3a216.

Reply:
Agree, that's even better. :+1: